lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -901,6 +901,21 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :allowed_from_server => false,
           :description => "Controls auto-instrumentation of resque at start up.  May be one of [auto|prepend|chain|disabled]."
         },
+        :'instrumentation.thread' => {
+          :default => 'auto',
+          :public => true,
+          :type => String,
+          :dynamic_name => true,
+          :allowed_from_server => false,
+          :description => "Controls auto-instrumentation of the Thread class at start up to enable accurate parent tracking for spans in applications manually inserting new relic tracing into threads. This does not enable the agent to automatically trace all threads created (see `instrumentation.thread.tracing`). May be one of [auto|prepend|chain|disabled]."
+        },
+        :'instrumentation.thread.tracing' => {
+          :default => false,
```

Comment:
I believe turning this on by default would be a breaking change (so needs a major release) for any customers who are creating multiple concurrent transactions using threads. This used to work because new relic could not see transactions across threads, but with this change a transaction will encompass all threads running in the process. We even have a test that confirmed this behavior works this way when thread tracing is off.

Reply:
I'm surprised to see this use a boolean value instead of the `auto|prepend|chain|disabled` options. Why did you decide to use a boolean?

Reply:
Since the way we install the instrumentation (which uses the prepend/chain stuff) is on the other config (`instrumentation.thread`), this is just controlling whether or not we insert ourselves into the thread when it's created.